File: code/remote.py
#!/usr/bin/python

#-----------------------------------#
# PROGRAM: SSH/SFTP Profile Manager #
# AUTHOR:  Sydney Ann Mason         #
# VERSION  20180111                 #
#-----------------------------------#

#---------#
# IMPORTS #
#---------#
import sys
import collections
import os

#---------------------------------------#
# FUNC: remote_action                   #
# DESC: Executes the specified SSH/SFTP #
#       protocol on the profile         #
# VARS:                                 #
#---------------------------------------#
def remote_action(cmd, prof):
    command = cmd + ' ' + prof
    os.system(command)

#--------------------------------------#
# FUNC: profile_maint                  #
# DESC: Adds a new profile to the list #
#       to the list of new profiles    #
# VARS:                                #
#--------------------------------------#
def profile_maint(url, prof):
    print(prof + ' = ' + url)

#------------------------------------#
# FUNC: print_profiles               #
# DESC: Prints the list of all saved #
#       profiles                     #
# VARS:                              #
#------------------------------------#
def print_profiles():
    print('Printing coming soon')

#---------------------------------#
# FUNC: help_menu                 #
# DESC: Prints executable actions #
# VARS:                           #
#   action_list, ordered dict of  #
#                actions          #
#---------------------------------#
def help_menu():
    action_list = collections.OrderedDict()

    action_list['ssh'] = 'executes SSH protocol on profile'
    action_list['sftp'] = 'executes SFTP protocol on profile'
    action_list['save'] = 'writes new profile to profile list'
    action_list['print'] = 'displays all currently saved profiles'
    action_list['help'] = 'displays all executable actions'

    print("#-----------#\n" +
          "# Help Menu #\n" +
          "#-----------#")

    for key, val in action_list.items():
        print(key + '\n\t' + val)

#----------------------------------#
# FUNC: remove                     #
# DESC: Deletes profile from the   #
#       list of the known profiles #
# VARS:                            #
#----------------------------------#
def remove(prof):
    print(prof)

#-------------------------------------#
# FUNC: main                          #
# DESC: Main function, takes CLI args #
#       and passes them to the        #
#       appropriate function          #
# VARS:                               #
#-------------------------------------#
def main(argv):
    try:
        if argv[0] == 'ssh' or argv[0] == 'sftp':
            remote_action(argv[0], argv[1])
        elif argv[0] == 'save':
            profile_maint(argv[1], argv[2])
        elif argv[0] == 'print':
            print_profiles()
        elif argv[0] == 'help':
            help_menu()
        elif argv[0] == 'remove':
            remove(prof)
    except Exception as e:
        print(e)

#--------------#
# CALL: main() #
#--------------#
if __name__ == "__main__":
    main(sys.argv[1:])

File: code/remote.cpp
/*******************************************
 * Remote, SSH/SFTP remote profile manager *
 * author: Sydney Ann Mason                *
 * version:                                *
 *******************************************/

/***********
 * Imports *
 ***********/
#include<iostream>
#include<vector>
#include<fstream>
#include<string>
#include<string.h>
#include<unistd.h>

using namespace std;

/**************************
 * Signature Declarations *
 **************************/
int main(int, char**);
int remote_action(char*, char*);
int add_profile(char*, char*);
int remove_profile(char*);
void print_profiles();
void help();

/******************
 * Function: Main *
 ******************/
int main(int argc, char* argv[]){
    try{
        vector<char*> arguments; //Vector holding CLI arguments

        //Reads argv into arguments
        for(int i = 1; i < argc; i++){
            arguments.push_back(argv[i]);
        }

        //Checks first argument to determine next course of action
        if(strcmp(arguments.at(1), "ssh") == 0 || strcmp(arguments.at(1), "sftp") == 0){
            remote_action(arguments.at(1), arguments.at(2));
        }
        else if(strcmp(arguments.at(1), "print") == 0){
            print_profiles();
        }
        else if(strcmp(arguments.at(1), "save") == 0){
            add_profile(arguments.at(2), arguments.at(3));
        }
        else if(strcmp(arguments.at(1), "remove") == 0){
            remove_profile(arguments.at(2));
        }
        else if(strcmp(arguments.at(1), "help") == 0){
            help();
        }
        else{
            cout << "Command:\n\t" << arguments.at(0) 
                 << endl << "is an invalid argument" 
                 << endl;
        }
    }
    catch(int ex){
        cout << "An err has occurred. Exception number " << ex << endl;
    }

    //Default return value on successful run
    return 0;
}

int remote_action(char* cmd, char* profile){
    return 0;
}

int add_profile(char* profileName, char* profileURL){
    return 0;
}

int remove_profile(char* profileName){
    return 0;
}

void print_profiles(){}

void help(){
    cout << "Help Menu:" << endl
         << "ssh\n\t" << "executes SSH protocol on given profile" << endl
         << "sftp\n\t" << "executes SFTP protocole on given profile" << endl
         << "print\n\t" << "prints all currently saved profiles" << endl
         << "save\n\t" << "writes new profile to the list of currently known profiles" << endl
         << "remove\n\t" << "removes specified profile from the list of known ones" << endl
         << "help\n\t" << "Displays help menu" << endl;
}
